client/dashboard/app/hooks/site-scan-state.ts

```diff
@@ -0,0 +1,34 @@
+import { SiteScan } from '@automattic/api-core';
```

Comment:
I think it is better to place this hook into the `sites/scan` directory; we can rename it too `use-scan-state.ts`, since it is only consumed by the main Scan page.

I'm also considering moving [app/hooks/site-backup-state.ts](https://github.com/Automattic/wp-calypso/blob/2b813fda922219981ba224c3fbc17c47556634ec/client/dashboard/app/hooks/site-backup-state.ts) to `sites/backup` since it is only used on `sites/backup-*` pages. This way we reduce the number of global hooks that are intended to work only on specific pages :)

Reply:
+1!

Reply:
Makes sense to me. I've put it there exactly because of the existing backup hook, but if we are moving it soon there's no point in adding this one there. I've moved the file.